cli/src/commands/extensions/permissions.rs

```diff
@@ -0,0 +1,112 @@
+use std::path::{Component, PathBuf};
+
+use anyhow::{anyhow, Result};
+use deno_runtime::permissions::PermissionsOptions;
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Default, Deserialize, Debug, Serialize)]
+pub struct Permissions {
+    read: Option<Vec<String>>,
+    write: Option<Vec<String>>,
+    run: Option<Vec<String>>,
+    net: Option<Vec<String>>,
+}
+
+// XXX In Deno, `Some(vec![])` actually means "allow all". We never
+// want that, so we manually convert those instances into `None` in the
+// getter methods below. We need to make sure to always go through these when
+// constructing a `PermissionsOptions` object.
+impl Permissions {
+    pub fn read(&self) -> Option<&Vec<String>> {
+        self.read.as_ref().and_then(|v| if v.is_empty() { None } else { Some(v) })
+    }
+
+    pub fn write(&self) -> Option<&Vec<String>> {
+        self.write.as_ref().and_then(|v| if v.is_empty() { None } else { Some(v) })
+    }
+
+    pub fn run(&self) -> Option<&Vec<String>> {
+        self.run.as_ref().and_then(|v| if v.is_empty() { None } else { Some(v) })
+    }
+
+    pub fn net(&self) -> Option<&Vec<String>> {
+        self.net.as_ref().and_then(|v| if v.is_empty() { None } else { Some(v) })
+    }
+
+    pub fn is_allow_none(&self) -> bool {
+        self.read().is_none()
+            && self.write().is_none()
+            && self.run().is_none()
+            && self.net().is_none()
+    }
+}
+
+impl TryFrom<&Permissions> for PermissionsOptions {
+    type Error = anyhow::Error;
+
+    fn try_from(value: &Permissions) -> Result<Self, Self::Error> {
+        let allow_read =
+            value.read().map(|read| read.iter().map(PathBuf::from).collect::<Vec<_>>());
+
+        let allow_write =
+            value.write().map(|write| write.iter().map(PathBuf::from).collect::<Vec<_>>());
+
+        let allow_net = value.net().cloned();
+        let allow_run = value.run().cloned();
+
+        Ok(PermissionsOptions {
+            allow_read,
+            allow_write,
+            allow_net,
+            allow_run,
+            ..Default::default()
```

Comment:
Other options fall back to default, but... there are no other options? This just seems like a footgun waiting to go off.

Reply:
Might be a remnant of the early attempts with only a couple of those permissions.